nestedtensor/csrc/functions.cpp

```diff
@@ -219,27 +218,56 @@ NestedTensor interpolate(NestedTensor input,
         throw error;
     }
 
+    TensorNode input_structure = input.get_structure();
     auto options = F::InterpolateFuncOptions().mode(int_mode);
-    if (size.has_value()) {
-      options = options.size(size.value().vec());
-    }
-    
-    if (scale_factor.has_value()) {
-      options = options.scale_factor(scale_factor.value().vec());
-    }
-
     if (align_corners.has_value()) {
       options.align_corners() = align_corners.value();
     }
 
-    TensorNode res = map(
+    // Either scale factor or size can be passed
```

Comment:
Is this true for torch.nn.functional.interpolate in general? It seems that the [source](https://pytorch.org/docs/master/_modules/torch/nn/functional.html#interpolate) passes down both arguments to the various kernels (e.g. upsample_linear1d or upsample_trilinear3d).

We don't need to resolve this for the purpose of this diff, but I think it's worth following up on even if just for the sake of throwing an error if we received a yet-to-be-supported (by us) combination or arguments.

Reply:
This is according to our documentation and very light testing that i did. i can look deeper into this if needed

Reply:
```
import torch 

a = torch.rand(5, 100, 100, 100)
print(a.size())

b = torch.nn.functional.interpolate(a, size=(50, 50))
print(b.size())

c = torch.nn.functional.interpolate(a, scale_factor=(1.1, 1.1))
print(c.size())

# ValueError: only one of size or scale_factor should be defined <<----------------
# d = torch.nn.functional.interpolate(a,  size=(50, 50), scale_factor=(1.1, 1.1))
# print(d.size())

```